identity/verifier.go

```diff
@@ -0,0 +1,27 @@
+package identity
+
+import (
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+type Verifier interface {
+	Verify(message []byte, signature []byte) bool
+}
+
+type ethereumVerifier struct {
+	peerIdentity Identity
+}
+
+func NewVerifier(peerIdentity Identity) *ethereumVerifier {
+	return &ethereumVerifier{peerIdentity}
+}
+
+func (ev *ethereumVerifier) Verify(message []byte, signature []byte) bool {
+	hash := messageHash(message)
+	pk, err := crypto.Ecrecover(hash, signature)
+	if err != nil {
+		return false
+	}
+	recoveredAddress := crypto.PubkeyToAddress(*crypto.ToECDSAPub(pk)).Hex()
+	return FromAddress(recoveredAddress) == ev.peerIdentity
```

Comment:
Just notice. Would be possible to drop attribute `peerIdentity`,
by changing interface:
`Verify(message []byte, signature []byte) bool`
->
`GetSigner(message []byte, signature []byte) Identity`

Reply:
I thought about this approach, but then verification from caller side would be a bit clumsy. Also comparing extracted identity with given identity would require a) have a reference to expected identity b) compare it everywhere verification is needed. I think bool return value is cleaner indication that verification succeeded or not